File: app/src/main/java/net/xcreen/restsms/AboutFragment.java
package net.xcreen.restsms;

import androidx.appcompat.widget.LinearLayoutCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.tabs.TabLayout;

public class AboutFragment extends Fragment {

    private TabLayout tabLayout;
    private ViewPager viewPager;
    private AppBarLayout mainAppBarLayout;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_about, container, false);

        //Set Tabs
        tabLayout = new TabLayout(getActivity());
        tabLayout.addTab(tabLayout.newTab().setText(getResources().getText(R.string.about_about_me)));
        tabLayout.addTab(tabLayout.newTab().setText(getResources().getText(R.string.about_app_information)));
        tabLayout.addTab(tabLayout.newTab().setText(getResources().getText(R.string.about_third_party_librarys)));
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
        tabLayout.setBackgroundColor(getResources().getColor(R.color.colorDarkBlack, null));
        tabLayout.setTabTextColors(getResources().getColor(R.color.colorWhite, null), getResources().getColor(R.color.colorLightBlue, null));
        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) { }

            @Override
            public void onTabReselected(TabLayout.Tab tab) { }
        });

        //Add TabLayout to the AppBarLayout
        mainAppBarLayout = getActivity().findViewById(R.id.appbar_layout);
        mainAppBarLayout.addView(tabLayout, new LinearLayoutCompat.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));

        //Set Page-Adapter to ViewPager
        FragmentPageAdapter pagerAdapter = new FragmentPageAdapter(getFragmentManager(), tabLayout.getTabCount());
        viewPager = rootView.findViewById(R.id.about_view_pager);
        viewPager.setAdapter(pagerAdapter);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));

        return rootView;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        //Remove Tab-Layout
        mainAppBarLayout.removeView(tabLayout);
    }
}

class FragmentPageAdapter extends FragmentStatePagerAdapter {
    private int numberOfTabs;

    public FragmentPageAdapter(FragmentManager fm, int numberOfTabs) {
        super(fm);
        this.numberOfTabs = numberOfTabs;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new AboutMeFragment();
            case 1:
                return new AboutMeFragment();
            case 2:
                return new AboutThirdPartyLibrarysFragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return numberOfTabs;
    }
}

File: app/src/main/java/net/xcreen/restsms/fragments/AboutFragment.kt
package net.xcreen.restsms.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.LinearLayoutCompat
import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.appbar.AppBarLayout
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
import com.google.android.material.tabs.TabLayoutMediator
import net.xcreen.restsms.R

class AboutFragment : Fragment() {
    private var tabLayout: TabLayout? = null
    private var viewPager: ViewPager2? = null
    private var mainAppBarLayout: AppBarLayout? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_about, container, false)
        if (activity != null) { //Set Tabs
            tabLayout = TabLayout(requireActivity())
            tabLayout!!.addTab(tabLayout!!.newTab()) // About-Me Tab
            tabLayout!!.addTab(tabLayout!!.newTab()) // App-Info Tab
            tabLayout!!.addTab(tabLayout!!.newTab()) // Third-Party-Info Tab
            tabLayout!!.tabGravity = TabLayout.GRAVITY_FILL
            tabLayout!!.setBackgroundColor(resources.getColor(R.color.colorDarkBlack, null))
            tabLayout!!.setTabTextColors(resources.getColor(R.color.colorWhite, null), resources.getColor(R.color.colorLightBlue, null))
            tabLayout!!.addOnTabSelectedListener(object : OnTabSelectedListener {
                override fun onTabSelected(tab: TabLayout.Tab) {
                    viewPager!!.currentItem = tab.position
                }
                override fun onTabUnselected(tab: TabLayout.Tab) {}
                override fun onTabReselected(tab: TabLayout.Tab) {}
            })
            //Add TabLayout to the AppBarLayout
            mainAppBarLayout = requireActivity().findViewById(R.id.appbar_layout)
            mainAppBarLayout!!.addView(tabLayout, LinearLayoutCompat.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT))
            //Set Page-Adapter to ViewPager
            val pagerAdapter = FragmentPageAdapter(this, tabLayout!!.tabCount)
            viewPager = rootView.findViewById(R.id.about_view_pager)
            viewPager!!.adapter = pagerAdapter
            TabLayoutMediator(tabLayout!!, viewPager!!) { tab, position ->
                tab.text = when (position) {
                    0 -> resources.getText(R.string.about_about_me)
                    1 -> resources.getText(R.string.about_app_information)
                    2 -> resources.getText(R.string.about_third_party_librarys)
                    else -> resources.getText(R.string.about_about_me)
                }
            }.attach()
        }
        return rootView
    }

    override fun onDestroyView() {
        super.onDestroyView()
        try {
            //Remove Tab-Layout
            mainAppBarLayout!!.removeView(tabLayout)
        }
        catch (ex: Exception){
            ex.printStackTrace()
        }
    }
}

internal class FragmentPageAdapter(fragment: Fragment, private val numberOfTabs: Int) : FragmentStateAdapter(fragment) {
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> AboutMeFragment()
            1 -> AboutAppFragment()
            2 -> AboutThirdPartyLibrarysFragment()
            else -> AboutMeFragment()
        }
    }

    override fun getItemCount(): Int {
        return numberOfTabs
    }
}